routeif.go

```diff
@@ -0,0 +1,5 @@
+package restful
+
+import "net/http"
+
+type RouteIfFunction func(httpRequest *http.Request) bool
```

Comment:
Did you consider using the wrapper type as argument ?
I know the detectRoute does not have this wrapper but what if the condition needs a path parameter value?

Reply:
I considered it but wasn't sure how to get / provide the wrapper object in that context, and since my conditions don't need them I didn't put a lot of time into it. How would one do that? Would it be much additional re-parsing?

Reply:
I had a quick look and get the feeling it will complicate/change the implementation considerably. Let's leave it like this.